### PR TITLE
Add proposed selection range support.

### PR DESCRIPTION
**Pull request by yaohaizh**

### CODE REVIEW

**Comment by yaohaizh** [comment]
@spoenemann

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 1]
```diff
@@ -1044,7 +1059,11 @@ class TextDocumentClientCapabilities {
 	 */
 	@Beta
 	CallHierarchyCapabilities callHierarchy
-
+	
+	/**
+	 * Capabilities specific to `textDocument/selectionRange` requests
+	 */
+	SelectionRangeCapabilities  selectionRange
```
Comment (spoenemann) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend]: Please add `@Beta` to this field, too.

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 2]
```diff
@@ -4885,4 +4911,56 @@ class CallHierarchySymbol {
 	@NonNull
 	Range selectionRange
 
+}
+
+@Beta
+@JsonRpcData
+class SelectionRangeParams {
```
Comment (spoenemann) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend]: Add class comment:
```
A parameter literal used in selection range requests.
```

File: org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend [hunk 3]
```diff
@@ -4885,4 +4911,56 @@ class CallHierarchySymbol {
...
+
+	new(@NonNull TextDocumentIdentifier textDocument, @NonNull List<Position> positions) {
+		this.textDocument = Preconditions.checkNotNull(textDocument, 'textDocument')
+		this.positions = Preconditions.checkNotNull(positions, 'positions')
+	}
+}
+
+@Beta
+@JsonRpcData
+class SelectionRange {
```
Comment (spoenemann) [inline org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend]: Add class comment:

```
A selection range represents a part of a selection hierarchy. A selection range
may have a parent selection range that contains it.
```

**Comment by spoenemann** [comment]
For reference: https://github.com/microsoft/vscode-languageserver-node/blob/625fbdbe492f7b4b95e6d29e0adf36c01b88d056/protocol/src/protocol.selectionRange.proposed.ts

**Comment by yaohaizh** [comment]
@spoenemann Updated the PR.

**Pull request merged**